src/MySqlConnector/Core/Row.cs

```diff
@@ -353,8 +353,15 @@ public double GetDouble(int ordinal)
 		public float GetFloat(int ordinal)
 		{
 			var value = GetValue(ordinal);
-			return value is decimal decimalValue ? (float) decimalValue :
-				(float) value;
+
+			// Loss of precision is expected, significant loss of information is not.
+			// Use explicit range checks to guard against that.
+			return value switch
+			{
+				double doubleValue => (doubleValue >= float.MinValue && doubleValue <= float.MaxValue ? (float) doubleValue : throw new InvalidCastException("The value cannot be safely cast to Single.")),
```

Comment:
This test could be if `doubleValue` is contained in any of the following ranges: `[float.MinValue, -float.Epsilon], [0, 0], [float.Epsilon, float.MaxValue]`; what do you think?

Reply:
So the difference would be, that double values between -/+ `float.Epsilon` and `0` would throw.

The question is, if it is expected that a `double` between `float.Epsilon` and `0` being converted to `0`. I think it is, because it's just a result of precision-loss.

But there could probably made a case against it as well, which would expect those values to end up as `float.Epsilon`. I think that is imprecise though, because then it would need be rounded to either `float.Epsilon` or `0`, whatever is closer. And I would assume, that the C#/ the processor is doing exactly that in the first place.

I would not throw in any of these cases, because this would be unexpected and a bit random.